pkg/broker/job_state_subscriber.go

```diff
@@ -44,6 +44,22 @@ func (jss *JobStateSubscriber) Notify(msg JobMsg) {
 			return
 		}
 	}
+	//TODO: Need to get the service instance
+	//TODO: NOTE: THIS NEEDS TO BREAK OUT TO OWN SUBSCRIBER
```

Comment:
I think if we have the time, this might be something we should do.

Reply:
I don't think it needs a different subscriber, but it could definitely use a helper function. It feels weird where it is now.

```golang

func (jss *JobStateSubscriber) handleDashboardURL(msg JobMsg) {
	// Update with dashboard URL.
	if msg.DashboardURL != "" {
		instance, err := jss.dao.GetServiceInstance(msg.InstanceUUID)
		if err != nil {
			log.Errorf("Error after job succeeded : %v", err)
			return
		}
		instance.DashboardURL = msg.DashboardURL
		err = jss.dao.SetServiceInstance(msg.InstanceUUID, instance)
		if err != nil {
			log.Errorf("Error after job succeeded : %v", err)
			return
		}
}
```

The `Notify` would then look cleaner:
```golang
// Notify external API to notify this subscriber of a change in the Job
func (jss *JobStateSubscriber) Notify(msg JobMsg) {
	log.Debugf("JobStateSubscriber Notify : msg state %v ", msg.State)
	id := msg.InstanceUUID
	if isBinding(msg) {
		id = msg.BindingUUID
	}
	if _, err := jss.dao.SetState(id, msg.State); err != nil {
		log.Errorf("Error JobStateSubscriber failed to set state after action %v completed with state %s err: %v", msg.State.Method, msg.State.State, err)
		return
	}
	if msg.State.State == apb.StateSucceeded {
		if err := jss.handleSucceeded(msg); err != nil {
			log.Errorf("Error after job succeeded : %v", err)
			return
		}
	}

	handleDashboardURL(msg)

	return
}
```